gravitee-apim-portal-webui-next/src/app/applications/application/application-tab-logs/application-log-messages/application-log-messages.harness.ts

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (C) 2025 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { ComponentHarness } from '@angular/cdk/testing';
+import { MatTableHarness } from '@angular/material/table/testing';
+
+import { PaginationHarness } from '../../../../../components/pagination/pagination.harness';
+
+export class ApplicationLogMessagesHarness extends ComponentHarness {
+  public static readonly hostSelector = 'app-application-log-messages';
+
+  protected locateTable = this.locatorFor(MatTableHarness);
+  protected locatePagination = this.locatorFor(PaginationHarness);
+
+  async getNumberOfRows(): Promise<number> {
+    return await this.locateTable()
+      .then(table => table.getRows())
+      .then(rows => rows.length);
+  }
+
+  isNoLogsMessageShown(): boolean {
+    const noLogsElement = this.getNoLogsMessage();
+    return !!noLogsElement && noLogsElement.innerHTML.includes('No message logs found.');
+  }
+
+  getNoLogsMessage(): Element | null {
+    return document.querySelector('.no-logs-found');
```

Comment:
To avoid using querySelector on ui particles we have created this https://github.com/gravitee-io/gravitee-ui-particles/blob/main/projects/ui-particles-angular/testing/harnesses/div.harness.ts. If you want to reuse it in portal next and add it in the test helpers for example